### PR TITLE
Added abbreviations for degree grades

### PR DESCRIPTION
**Pull request by alaric-dfe**
This is a request from analytics folks who want to display degree grades, but don't want massive long names like "Upper Second Class Honours (2:1)"

### CODE REVIEW

File: lib/dfe/reference_data/degrees/grades.rb [hunk 1]
```diff
@@ -15,6 +16,7 @@ module Degrees
           '8741765a-13d8-4550-a413-c5a860a59d25' =>
           { hesa_code: '01',
             name: 'First-class honours',
+            abbreviation: '1',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:19]:
```suggestion
            abbreviation: 'First',
```

I don't think `1` makes sense as an abbreviation.

File: lib/dfe/reference_data/degrees/grades.rb [hunk 2]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
           { hesa_code: '04',
             name: 'Undivided second-class honours',
+            abbreviation: '2',
             suggestion_synonyms: [],
             match_synonyms: ['Undivided second class honours'],
             group: :other },
           '3d57e5f7-7904-4895-a982-ed953fff3f94' =>
           { hesa_code: '05',
             name: 'Third-class honours',
+            abbreviation: '3',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:52]:
```suggestion
            abbreviation: 'Third',
```

File: lib/dfe/reference_data/degrees/grades.rb [hunk 3]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
             suggestion_synonyms: [],
             match_synonyms: ['Third class honours', 'Third'],
             group: :main_undergrad },
           'da6f57ae-7de0-4215-82df-52322129e91f' =>
           { hesa_code: '06',
             name: 'Fourth-class honours',
+            abbreviation: '4',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:59]:
```suggestion
            abbreviation: 'Fourth',
```

File: lib/dfe/reference_data/degrees/grades.rb [hunk 4]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
           { hesa_code: '08',
             name: 'Aegrotat',
+            abbreviation: 'Aegrotat',
             suggestion_synonyms: [],
             match_synonyms: [],
             group: :other },
           '4d3cc9d3-81dd-4457-ae2b-ee0bd298c51e' =>
           { hesa_code: '10',
             name: 'Ordinary degree',
+            abbreviation: 'Ordinary',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:80]:
```suggestion
            abbreviation: 'Ordinary',
```

I'm not sure this makes sense as an abbreviation.

File: lib/dfe/reference_data/degrees/grades.rb [hunk 5]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
             suggestion_synonyms: [],
             match_synonyms: ['Ordinary'],
             group: :other  },
           '2db26c4f-59d9-4431-8c92-4396d8e35b2d' =>
           { hesa_code: '11',
             name: 'General degree',
+            abbreviation: 'General',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:87]: I don't think this makes sense as an abbreviation.

**Review by edwardhorsford — CHANGES_REQUESTED** [review]
I don't think several of these make sense as abbreviations.

If they need to be these specific values, because a single team needs them, then perhaps they should be named something else.

File: lib/dfe/reference_data/degrees/grades.rb [hunk 6]
```diff
@@ -26,66 +28,77 @@ module Degrees
           'e2fe18d4-8655-47cf-ab1a-8c3e0b0f078f' =>
           { hesa_code: '02',
             name: 'Upper second-class honours (2:1)',
+            abbreviation: '2:1',
```
Comment (alexchristensenedu) [inline lib/dfe/reference_data/degrees/grades.rb:31]: Agree with Ed's points, can this also be changed to 'Upper second' so it's consistent with the other abbreviations

File: lib/dfe/reference_data/degrees/grades.rb [hunk 7]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
             suggestion_synonyms: [],
             match_synonyms: ['Upper second class honours', '2:1', '2.1'],
             group: :main_undergrad },
           '377a46ea-d6c6-4e87-9728-c1f0dd0ef109' =>
           { hesa_code: '03',
             name: 'Lower second-class honours (2:2)',
+            abbreviation: '2:2',
```
Comment (alexchristensenedu) [inline lib/dfe/reference_data/degrees/grades.rb:38]: Can this also be changed to 'Lower second' so it's consistent with the other abbreviations

File: lib/dfe/reference_data/degrees/grades.rb [hunk 8]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
             suggestion_synonyms: [],
             match_synonyms: ['Lower second class honours', '2:2', '2.2'],
             group: :main_undergrad },
           '7f08cd7c-96d2-4706-9148-32468a6d67f9' =>
           { hesa_code: '04',
             name: 'Undivided second-class honours',
+            abbreviation: '2',
```
Comment (alexchristensenedu) [inline lib/dfe/reference_data/degrees/grades.rb:45]: Can this also be changed to 'Undivided second' so it's consistent with the other abbreviations

**Review by alexchristensenedu — CHANGES_REQUESTED** [review]
Ed's point is fair. Added comments

**Comment by edwardhorsford** [comment]
I wonder if this should be `short_name`? they're not really abbreviations.

File: lib/dfe/reference_data/degrees/grades.rb (hunk 4 above)
Comment (alaric-dfe) [reply lib/dfe/reference_data/degrees/grades.rb]: Does it make more sense now as a short_name?

File: lib/dfe/reference_data/degrees/grades.rb (hunk 5 above)
Comment (alaric-dfe) [reply lib/dfe/reference_data/degrees/grades.rb]: Does it make more sense now as a short_name?

File: lib/dfe/reference_data/degrees/grades.rb [hunk 9]
```diff
@@ -118,6 +134,7 @@ module Degrees
         field_descriptions: {
           id: 'A unique identifier for the grade',
           name: 'The long name of this grade, eg "First-class honours"',
+          abbreviation: 'The short name of this grade, eg "2:2"',
```
Comment (alexchristensenedu) [inline lib/dfe/reference_data/degrees/grades.rb:137]: this needs to change to 'short_name'

**Review by alexchristensenedu — CHANGES_REQUESTED** [review]

File: lib/dfe/reference_data/degrees/grades.rb (hunk 5 above)
Comment (edwardhorsford) [reply lib/dfe/reference_data/degrees/grades.rb]: Still not sure! it's such a generic word!

File: lib/dfe/reference_data/degrees/grades.rb [hunk 10]
```diff
@@ -118,6 +134,7 @@ module Degrees
         field_descriptions: {
           id: 'A unique identifier for the grade',
           name: 'The long name of this grade, eg "First-class honours"',
+          short_name: 'The short name of this grade, eg "2:2"',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:137]: I think this example needs updating.

File: lib/dfe/reference_data/degrees/grades.rb [hunk 11]
```diff
@@ -26,66 +28,77 @@ module Degrees
           'e2fe18d4-8655-47cf-ab1a-8c3e0b0f078f' =>
           { hesa_code: '02',
             name: 'Upper second-class honours (2:1)',
+            short_name: 'Upper Second',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:31]:
```suggestion
            short_name: 'Upper second',
```

File: lib/dfe/reference_data/degrees/grades.rb [hunk 12]
```diff
@@ -26,66 +28,77 @@ module Degrees
...
             suggestion_synonyms: [],
             match_synonyms: ['Upper second class honours', '2:1', '2.1'],
             group: :main_undergrad },
           '377a46ea-d6c6-4e87-9728-c1f0dd0ef109' =>
           { hesa_code: '03',
             name: 'Lower second-class honours (2:2)',
+            short_name: 'Lower Second',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/degrees/grades.rb:38]:
```suggestion
            short_name: 'Lower second',
```

**Review by alexchristensenedu — APPROVED** [review]
Approved on my side

**Pull request merged**